include/dlaf_c/factorization/cholesky.h

```diff
@@ -0,0 +1,83 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2023, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#pragma once
+
+/// @file
+
+#include <dlaf_c/desc.h>
+#include <dlaf_c/utils.h>
+
+#ifdef DLAF_WITH_SCALAPACK
+
+/// Cholesky decomposition
+///
+/// @remark This function is available only when DLAF_WITH_SCALAPACK=ON.
+///
+/// @pre The matrix \f$\mathbf{A}\f$ is assumed to be distributed and in host memory. Moving to and from
+/// GPU memory is handled internally.
+///
+/// @pre Submatrices are currently not supported, so @p n is the size of the full matrix
+/// \f$\mathbf{A}\f$ and @p ia, and @p ja need to be 1.
+///
+/// @post The pika runtime is resumed when this function is called and suspended when the call
+/// terminates.
+///
+/// @param uplo indicates wheather the upper ('U') or lower ('L') triangular part of the global submatrix
+/// \f$\mathbf{A}\f$ is referenced
+/// @param n order of the sumbatrix \f$\mathbf{A}\f$ used in the computation
+/// @param a Local part of the global matrix \f$\mathbf{A}\f$
+/// @param ia row index of the global matrix \f$\mathbf{A}\f$ identifying the first row of the submatrix
+/// \f$\mathbf{A}\f$, has to be 1
+/// @param ja column index of the global matrix \f$\mathbf{A}\f$ identifying the first column of the
+/// submatrix \f$\mathbf{A}\f$, has to be 1
+/// @param desca ScaLAPACK array descriptor of the global matrix \f$\mathbf{A}\f$
+/// @param info 0 if the factorization completed normally
```

Comment:
I'm not sure we can do anything about this already now, but I guess `info` will always be `0` or the algorithm will terminate? We don't currently have a way to handle all errors in algorithms and report them to callers (only those happening at schedule time).

One potential direction for this would be to wrap algorithm execution in something like an `async_scope` and then explicitly wait for all work in the `async_scope` (and check errors) before returning from the C API. This probably deserves its own issue so that we don't forget about it.

No action required for this PR.